### PR TITLE
#163284045 Users update profile feature

### PR DESCRIPTION
**Pull request by julietezekwe**
#### What does this PR do?
This PR implements user profile update

#### Description of Task to be completed?
- Logged user can update their profile
- Input for the profile update is validated
- Test coverage for the added feature

#### How should this be manually tested?
-Login to the application and visit `http://localhost:9000/api/v1/users/profile`
- Input the required fields to update profile

#### Any background context you want to provide?
Required fields includes `bio`, `imgURL`, `userType`.

#### What are the relevant pivotal tracker stories?
[#163284045](https://www.pivotaltracker.com/story/show/163284045)

#### Screenshots (if appropriate)
None

#### Questions:
None

### CODE REVIEW

**Review by abejide001 — APPROVED** [review]
Looks good to me

**Review by wombolo — APPROVED** [review]
LGTM

**Review by daniellamarr — APPROVED** [review]
This is okay

**Review by Proception — COMMENTED** [review]
Looks good to me

**Review by Proception — APPROVED** [review]

**Comment by julietezekwe** [comment]
This test drop is in no way relating to this feature, I made sure all passing tests are still passing and more.

File: src/controllers/UsersController.js [hunk 1]
```diff
@@ -344,7 +346,56 @@ class UsersController {
    * @param {object} req
    * @param {object} res
    * @memberof UsersController
-   * @returns successful pasword reset
+   * @returns successful profile update
+   */
+  static async updateProfile(req, res) {
```
Comment (andela-dbamidele) [inline src/controllers/UsersController.js:351]: The function name should compliment the url in the description. It's creating some confusion.

File: src/middlewares/ParamsChecker.js [hunk 2]
```diff
@@ -26,8 +26,8 @@ class ParamsChecker {
       }
       return next();
     };
-    if (userId) checkParam(userId);
-    if (artId) checkParam(artId);
+    /* istanbul ignore next */if (userId) checkParam(userId);
```
Comment (andela-dbamidele) [inline src/middlewares/ParamsChecker.js]: Why this?

File: test/controllers/commentControllerTest.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ describe('Comments Endpoint API Test', () => {
       });
   });
   describe('Comments POST REQUESTS', () => {
-    it('it should add a product that exists', (done) => {
+    it('it should comment add comment for a product that exists', (done) => {
```
Comment (andela-dbamidele) [inline test/controllers/commentControllerTest.js]: Typo in the description

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/controllers/UsersController.js (hunk 1 above)
Comment (julietezekwe) [reply src/controllers/UsersController.js:351]: Fixed

**Review by andela-dbamidele — APPROVED** [review]

**Pull request merged**